include/libcyphal/transport/udp/udp_transport_impl.hpp

```diff
@@ -175,6 +176,9 @@ class TransportImpl final : private TransportDelegate, public IUdpTransport  //
                   MediaArray&&           media_array)
         : TransportDelegate{memory_resources}
         , media_array_{std::move(media_array)}
+        , msg_rx_session_nodes_{memory_resources.general}
+        , svc_request_rx_session_nodes_{memory_resources.general}
+        , svc_response_rx_session_nodes_{memory_resources.general}
```

Comment:
Currently, it might seems that we can go with just one combined tree, but I anticipate that their node types might deviate from each other (although currently they are exactly the same - all derived from `BaseNode`; without their own specifics). I'll keep in mind these 3 trees and if needed will make proper changes.

Reply:
As we discussed earlier today, we should expect them to be different indeed; plus, having separate trees reduces their depth which improves WCET